src/ServiceControl.Monitoring/Timings/IntervalsStore.cs

```diff
@@ -176,5 +178,28 @@ public class TimeInterval
         internal const int NumberOfHistoricalIntervals = 4 * 5;
 
         static TimeSpan IntervalSize = TimeSpan.FromSeconds(15);
+
+        IDictionary<string, IEnumerable<EndpointInstanceId>> IKnowAboutEndpoints.AllEndpointData()
+        {
+            return intervals.Keys
+                .GroupBy(i => i.EndpointName)
+                .ToDictionary(g => g.Key, g => g.AsEnumerable());
+        }
+
+        protected static MonitoredEndpointValues AggregateTimings(List<EndpointInstanceIntervals> timings)
```

Comment:
Either that or rename `AggregateRetries` to `AggregateToCount` and push it down into the timing store. I think we'll want to refactor some of this afterwards anyway.